contracts/clients-tests/BlsSigner.test.ts

```diff
@@ -357,6 +367,165 @@ describe("BlsSigner", () => {
     expect(signedMessage).to.deep.equal(expectedSignature);
     expect(RLP.decode(signedMessage)).to.deep.equal(blsWalletSignerSignature);
   });
+
+  it("should get the balance of an account", async () => {
+    // Arrange
+    const expectedBalance = await regularProvider.getBalance(
+      blsSigner.wallet.address,
+    );
+
+    // Act
+    const result = await blsSigner.getBalance();
+
+    // Assert
+    expect(result).to.equal(expectedBalance);
+  });
+
+  it("should get the balance of an account at specific block height", async () => {
+    // Arrange
+    const expectedBalance = parseEther("0");
+
+    // Act
+    const result = await blsSigner.getBalance("earliest");
+
+    // Assert
+    expect(result).to.equal(expectedBalance);
+  });
+
+  it("should get the chain id the wallet is connected to", async () => {
+    // Arrange
+    const expectedChainId = 31337;
```

Comment:
I don't think these tests will ever be run against a non-hardhat network, but you could pull the `chainId` from the regular provider just in case. https://docs.ethers.io/v5/api/providers/provider/#Provider-getNetwork
```suggestion
    const { chainId: expectedChainId } = await regularProvider.getNetwork();
```

Reply:
> Some minor comments, but overall looks good.
> 
> Commit history is a bit...much given we squashed the last one and this includes some of those old commits. You may be able to [git cherry-pick](https://git-scm.com/docs/git-cherry-pick) your commits/changes for this PR into a clean state from `main`. I can help with that offline.

Yeah I tried (and failed) rebasing. I'll give cherry picking a go and if I get nowhere, help offline sounds good :)

Reply:
Good shout, updated